packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts

```diff
@@ -117,15 +114,15 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
           targetIndex: `${stateP.indexPrefix}_${stateP.kibanaVersion}_001`,
           targetIndexMappings: mergeMigrationMappingPropertyHashes(
             stateP.targetIndexMappings,
-            indices[aliases[stateP.currentAlias]].mappings
+            indices[aliases[stateP.currentAlias]!].mappings
```

Comment:
NIT: if we're `!` forcecasting all usages, not sure the changed from `Record<string, string>` to `<string, string | undefined>` was really useful?

Reply:
yeah we didn't actually increase the type safety. But the `!` is at least a warning that we should check to make sure it's safe.

It's frustrating that TS doesn't narrow the type without first assigning it to a variable e.g.:
```
if (
  aliases[stateP.currentAlias] != null
) {
  // aliases[stateP.currentAlias] should be types as string but is still string | undefined
}
```

```
let currentIndex = aliases[stateP.currentAlias];
if (
  currentIndex != null
) {
  // currentIndex correctly narrowed as string
}
```